test/e2e/service.go

```diff
@@ -966,6 +968,99 @@ var _ = framework.KubeDescribe("Services", func() {
 			framework.Failf("expected un-ready endpoint for Service %v within %v, stdout: %v", t.name, kubeProxyLagTimeout, stdout)
 		}
 	})
+
+	It("should be able to create services of type LoadBalancer and externalTraffic=localOnly [Slow]", func() {
```

Comment:
The test itself should, at the very least, check if the feature is turned
on and skip if it isn't.

On Mon, Aug 22, 2016 at 3:47 PM, Tim Hockin <notifications@github.com>
wrote:

> In test/e2e/service.go
> <https://github.com/kubernetes/kubernetes/pull/29409#discussion_r75773427>
> :
>
> > @@ -966,6 +968,99 @@ var _ = framework.KubeDescribe("Services", func() {
> >  			framework.Failf("expected un-ready endpoint for Service %v within %v, stdout: %v", t.name, kubeProxyLagTimeout, stdout)
> >  		}
> >  	})
> > +
> > +	It("should be able to create services of type LoadBalancer and externalTraffic=localOnly [Slow]", func() {
>
> But without the feature-gate flag set, it will ALWAYS fail On Mon, Aug 22,
> 2016 at 3:44 PM, Daniel Smith <notifications@github.com> wrote:
> … <#m_-7807339988206321156_>
> In test/e2e/service.go <#29409 (comment)
> <https://github.com/kubernetes/kubernetes/pull/29409#discussion_r75772942>>
> : > @@ -966,6 +968,99 @@ var _ = framework.KubeDescribe("Services",
> func() { > framework.Failf("expected un-ready endpoint for Service %v
> within %v, stdout: %v", t.name, kubeProxyLagTimeout, stdout) > } > }) > +
> > + It("should be able to create services of type LoadBalancer and
> externalTraffic=localOnly [Slow]", func() { It's marked as slow, it doesn't
> run on presubmits. You're clear. … <#m_-8652281336046335949_> On Sun, Aug
> 21, 2016 at 10:22 PM, Tim Hockin ***@***.***> wrote: In test/e2e/service.go
> <#29409 <https://github.com/kubernetes/kubernetes/pull/29409> (comment) <#29409
> (comment)
> <https://github.com/kubernetes/kubernetes/pull/29409#discussion_r75619413>>>
> : > @@ -966,6 +968,99 @@ var _ = framework.KubeDescribe("Services",
> func() { > framework.Failf("expected un-ready endpoint for Service %v
> within %v, stdout: %v", t.name, kubeProxyLagTimeout, stdout) > } > }) > +
> > + It("should be able to create services of type LoadBalancer and
> externalTraffic=localOnly [Slow]", func() { @lavalamp
> <https://github.com/lavalamp> <https://github.com/lavalamp> <
> https://github.com/lavalamp> @ixdy <https://github.com/ixdy> <
> https://github.com/ixdy> <https://github.com/ixdy> This is going to slow
> down e2e runs. We have almost all the load-balancer testing in one case,
> but this is fundamentally a different test and can't be combined. Just FYI
> — You are receiving this because you were mentioned. Reply to this email
> directly, view it on GitHub <https://github.com/
> kubernetes/kubernetes#29409
> <https://github.com/kubernetes/kubernetes/pull/29409>/files/f5f99257867d769d6b8246d95ae9cb
> 0cf139d608..59883dc
> <https://github.com/kubernetes/kubernetes/commit/59883dc63dbea4bc31ef2ef85d3274a0ea383e78>
> #r75619413>, or mute the thread <https://github.com/
> notifications/unsubscribe- auth/AAngll5tbz0nqN0yh-z6b-
> Xxacpk29lnks5qiTIFgaJpZM4JSUXH> . — You are receiving this because you
> were mentioned. Reply to this email directly, view it on GitHub <
> https://github.com/kubernetes/kubernetes/pull/29409/files/
> f5f99257867d769d6b8246d95ae9cb0cf139d608..59883dc63dbea4bc31ef2ef85d3274
> a0ea383e78#r75772942>, or mute the thread <https://github.com/
> notifications/unsubscribe-auth/AFVgVC4N63gZn1pHicjxIf4vnLU1Ia
> sBks5qiiZpgaJpZM4JSUXH> .
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29409/files/f5f99257867d769d6b8246d95ae9cb0cf139d608..59883dc63dbea4bc31ef2ef85d3274a0ea383e78#r75773427>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnglhcY1ccgiMmeShP4VYoPyEPjuEIpks5qiicRgaJpZM4JSUXH>
> .
>